### DISCUSSION TITLE
[Tooltip] How to sync TooltipProvider in different React trees?

### DISCUSSION THREAD

**Discussion by ivanbanov** [radix-ui/primitives#1757] [Ideas]
Hi, we have a big old code base written in angular, in the process of migrating it to React we are also applying our design-system components which may end up in different react trees, at least for now. The problem is that we have no single point to set the `TooltipProvider`, it would be great if the `TooltipProvider` could listen to a singleton instead of a React context.

What would you suggest instead?

**Comment by benoitgrelard** [comment]
I think the use-case is fringe enough that we're unlikely to make any changes for this.

However, depending on where your sub-trees are, it probably largely doesn't matter as it's mostly to ensure things like being able to go quickly from one tooltip to another without having to wait for the delay. If those react sub-trees are far away that's unlikely anyway.

I'd recommend putting the `TooltipProvider` at the root of each sub-tree.
